app/models/rating_vote.rb

```diff
@@ -2,28 +2,24 @@ class RatingVote < ApplicationRecord
   belongs_to :article
   belongs_to :user
 
-  validates :user_id, uniqueness: { scope: :article_id }
+  validates :user_id, uniqueness: { scope: %i[article_id context] }
   validates :group, inclusion: { in: %w[experience_level] }
+  validates :context, inclusion: { in: %w[explicit readinglist_reaction comment] }
   validates :rating, numericality: { greater_than: 0.0, less_than_or_equal_to: 10.0 }
   validate :permissions
 
+  after_create_commit :assign_article_rating
+
   counter_culture :article
   counter_culture :user
 
   def assign_article_rating
-    ratings = article.rating_votes.where(group: group).pluck(:rating)
-    average = ratings.sum / ratings.size
-
-    article.update_columns(
-      experience_level_rating: average,
-      experience_level_rating_distribution: ratings.max - ratings.min,
-      last_experience_level_rating_at: Time.current,
-    )
+    RatingVotes::AssignRatingWorker.perform_async(article_id)
```

Comment:
I think we want a guard clause here to only run this worker is `category == "readinglist" && current_user.experience_level`. We have that clause in the reactions controller but it looks like there are other ways to create a RatingVote which means we should probably add that clause here as well.

Reply:
We want to run this any time a rating vote is successfully created, which is currently only of group `experience_level` so I think it's fine here. We're still accepting of explicit rating vote creation.

Reply:
> We want to run this any time a rating vote is successfully created, which is currently only of group experience_level

Right but this creates a worker for EVERY rating vote, not just `experience_level`

Reply:
Currently the only type of rating vote that exists is the `experience_level` type, and if we added other types, we'd probably want still run `AssignRatingWorker` after a new one is created, maybe an update to different columns, but right now I think this works.